background.js

```diff
@@ -1,24 +1,27 @@
 function onBeforeRequest(data){
     //perhaps put 'data' in parameter
     console.log("top of on before request method");
-    console.log(data);
+    console.log(data.initiator);
     //need to return something
     //this is where blocking goes and also figuring out whether something needs to be blocked
-    var urlNotFixed = window.location.toString();
+
+    //rework all of this!
+    var urlNotFixed = data.initiator;
     urlNotFixed.replace(" ", "");
     urlNotFixed.toLowerCase();
     var urlFixed = urlNotFixed;
-    console.log(urlFixed);
-    var blockedList = "facebook.com";
-    if(urlFixed.contains(blockedList)){
-        return webRequest.BlockingResponse;
+    if(urlFixed.includes(blockedList)){
+        console.log("inside if statement");
+        //okay so it's entering the if statement correctly but then nothing happens and the code continues to run
+        return {cancel : true};
     }
 
     //for block list: look at string mdn on google and look at the methods for the string object
     
 }
+var blockedList = "facebook.com";
```

Comment:
Best practice these days is to use `const` whenever possible and `let` when it's not. In this case, since we don't need to modify the value of the variable after we initialize it, we can and should use `const`.

...Why? Basically, because the `const` keyword imposes a bunch of constraints on how and where the variable can be used, and these constraints make it harder to introduce bugs. For example: if you declare a variable with `var`, JavaScript will not complain if you accidentally use it before initializing it with a value, which will probably lead to unexpected behavior since the default value probably won't be what you wanted or intended. When you use `const`, you must initialize at the same time that you declare, so the language just straight up won't let you make the mistake of using an uninitialized `const`.
 
Some more detail here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const and here: https://ponyfoo.com/articles/var-let-const